Text/HTML/Parser.hs

```diff
@@ -106,7 +106,7 @@ tagOpen =
 -- TODO: This isn't right
 endTagOpen :: Parser Token
 endTagOpen = do
-    name <- takeWhile $ \c -> isAsciiUpper c || isAsciiLower c
+    name <- takeWhile $ notInClass "\x09\x0a\x0c />"
```

Comment:
Hmm, this is quite odd; at first glance the [specification](http://w3c.github.io/html/syntax.html#end-tag-open-state) seems rather explicit here but the `TODO` above suggests that I approximated here. I'm afraid this is all very much paged out at this point. Do you have any insight into what is happening here?